x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1019,4 +1041,15 @@ static long machineMemoryFromStats(OsStats stats) {
             new SystemIndexDescriptor(InferenceIndexConstants.INDEX_PATTERN, "Contains ML model configuration and statistics")
         );
     }
+
+    @Override
+    public Collection<AutoscalingDeciderService<? extends AutoscalingDecider>> deciders() {
+        logger.info("In ML deciders");
+        if (enabled) {
+            assert memoryTracker.get() != null;
+            return List.of(new MlAutoscalingDeciderService(memoryTracker.get()));
+        } else {
+            return List.of();
+        }
+    }
```

Comment:
`MachineLearning` can share the `memoryTracker` to the it's decider service and make the service available to the `Autoscaling` framework.